src/components/NcRichText/NcRichText.vue

```diff
@@ -387,6 +387,13 @@ export default {
 	},
 
 	methods: {
+		handleReferenceListLoaded(oldHeight) {
+			this.$nextTick(() => {
+				// emit height difference to parent component
+				const heightDiff = this.$refs.referenceList.$el.offsetHeight - oldHeight
+				this.$emit('reference-list-loaded', heightDiff)
+			})
+		},
```

Comment:
```suggestion
		handleReferenceListLoaded() {
			const oldHeight = this.$refs.referenceList.$el.offsetHeight
			this.$nextTick(() => {
				// emit height difference to parent component
				const heightDiff = this.$refs.referenceList.$el.offsetHeight - oldHeight
				this.$emit('reference-list-loaded', heightDiff)
			})
		},
```

Reply:
But in general would it not just work if you emit any event here and in Talk react on that element by scrolling the element into view?
Then you do not need any height calculation here

Reply:
calculating the old height here may be already updated with the reference data ? 

> in Talk react on that element by scrolling the element into view?

The goal not to scroll to the element but to calculate the extra height added (it will put the view above its previous position) and then we scroll back to the right position.

Reply:
> calculating the old height here may be already updated with the reference data added height ?

I am not sure about the timing of Vue's event handling, sadly I could not find good documentation about this. Maybe @ShGKme knows more?

> The goal not to scroll to the element but to calculate the extra height added (it will scroll the view above its previous position) and then we scroll back to the right position.

I do not know the richtext part good enough (and the talk code) so your solution might be the best.
But then I still would not pass the height as magic number to the event but use a context object with proper property naming like

`this.$emit('reference-list-loaded', { heightDiff })`